File: src/test/java/at/fhhagenberg/sqe/api/BuildingFloorServiceTest.java
package at.fhhagenberg.sqe.api;

import at.fhhagenberg.sqe.di.DI;
import at.fhhagenberg.sqe.entity.BuildingFloor;
import com.google.inject.Injector;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;
import java.rmi.RemoteException;
import java.util.List;
import static org.junit.jupiter.api.Assertions.*;

public class BuildingFloorServiceTest{

    private BuildingFloorService service;

    @BeforeEach
    public void setUp() {
        Injector injector = DI.getInjector();
        service = injector.getInstance(BuildingFloorService.class);
    }

    @Test
    public void testGetAll() throws RemoteException {
        List<BuildingFloor> floors = service.getAll();

        assertNotNull(floors);
        assertEquals(2, floors.size());
    }

    @Nested
    class GetTest {
        @Test
        public void testGetRegular() throws RemoteException {
            BuildingFloor floor = service.get(1);

            assertNotNull(floor);
            assertEquals(floor.getFloorNumber(), 1);
        }

        @Test
        public void testGetInvalidFloorNumber() throws RemoteException {
            BuildingFloor floor = service.get(-1);

            assertNull(floor);
        }
    }
}

File: src/test/kotlin/at/fhhagenberg/sqe/api/BuildingFloorServiceTest.kt
package at.fhhagenberg.sqe.api

import at.fhhagenberg.sqe.di.TestDI
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.Mockito
import sqelevator.ConnectableIElevator
import sqelevator.IElevator
import java.rmi.RemoteException
import kotlin.test.*

class BuildingFloorServiceTest {

    private lateinit var service: BuildingFloorService
    private lateinit var realIElevator: IElevator

    @BeforeEach
    fun setUp() {
        val injector = TestDI.createMockInjector()
        service = injector.getInstance(BuildingFloorService::class.java)
        realIElevator = injector.getInstance(ConnectableIElevator::class.java)
    }

    @Test
    @Throws(RemoteException::class)
    fun testGetAll() {
        val floors = service.getAll()

        assertEquals(2, floors.size)
    }

    @Test
    @Throws(RemoteException::class)
    fun testGetAllNegative() {
        Mockito.`when`(realIElevator.floorNum).thenReturn(-1)
        val floors = service.getAll()

        assertEquals(0, floors.size)
    }

    @Test
    @Throws(RemoteException::class)
    fun testGet() {
        val floor = service.get(1)!!

        assertEquals(floor.floorNumber, 1)
        assertEquals(floor.isDownActive, false)
        assertEquals(floor.isUpActive, false)
    }

    @Test
    @Throws(RemoteException::class)
    fun testGetInvalidFloorNumber() {
        val floor = service.get(-1)

        assertNull(floor)
    }
}